File: hackerrank/makingAnagrams.js
// Complete the makeAnagram function in the editor below. It must return an integer representing the minimum total characters that must be deleted to make the strings anagrams.

function makeAnagram(a, b) {
  let deletions = 0;
  let letterFreqs = {};

  for (let char of a) {
    if (letterFreqs[char]) {
      letterFreqs[char] += 1;
    } else {
      letterFreqs[char] = 1;
    }
  }

  for (let char of b) {
    if (letterFreqs[char]) {
      letterFreqs[char] -= 1;
    } else {
      letterFreqs[char] = -1;
    }
  }

  for (let freq in letterFreqs) {
    deletions += Math.abs(letterFreqs[freq]);
  }
  return deletions;
}

console.log(makeAnagram("cde", "abc")); // should return 4
console.log(
  makeAnagram("fcrxzwscanmligyxyvym", "jxwtrhvujlmrpdoqbisbwhmgpmeoke")
); // should return 30
console.log(makeAnagram("showman", "woman")); // should return 2

File: hackerrank/Python/makingAnagrams.py
# Complete the makeAnagram function in the editor below. It must return an integer representing the minimum total characters that must be deleted to make the strings anagrams.


def makeAnagram(a, b):
    deletions = 0
    letterFreqs = {}

    for char in a:
        if char not in letterFreqs:
            letterFreqs[char] = 1
        else:
            letterFreqs[char] += 1

    for char in b:
        if char not in letterFreqs:
            letterFreqs[char] = -1
        else:
            letterFreqs[char] -= 1

    for freq in letterFreqs:
        deletions += abs(letterFreqs[freq])

    return deletions


print(makeAnagram("cde", "abc"))  # should return 4
# should return 30
print(makeAnagram("fcrxzwscanmligyxyvym", "jxwtrhvujlmrpdoqbisbwhmgpmeoke"))
print(makeAnagram("showman", "woman"))  # should return 2
